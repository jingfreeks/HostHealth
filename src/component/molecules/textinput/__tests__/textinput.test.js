import React from 'react';
import {render,fireEvent} from '@testing-library/react-native';
import TextInput from '../textinput';

describe('Text Input Component', () => {
  it('Should work as expected to get snapshot', () => {
    const all = render(<TextInput />);
    expect(all.toJSON()).toMatchSnapshot();
  });
  it('Should work as expected to set text type to password to get snapshot', () => {
    const all = render(<TextInput type="password"/>);
    expect(all.toJSON()).toMatchSnapshot();
  });
  it('Should work as expected to trigger show password when press the visible icon', () => {
    const all = render(<TextInput type="password"/>);
    const el = all.getByTestId('MaterialsIconShowPasswordTestId');
    fireEvent(el, 'onPress');
    expect(all.toJSON()).toBeTruthy();
  });
});
