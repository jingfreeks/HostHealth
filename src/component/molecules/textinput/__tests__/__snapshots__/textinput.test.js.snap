// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Input Component Should work as expected to get snapshot 1`] = `
<RCTSafeAreaView
  style={
    [
      {},
    ]
  }
>
  <View
    alignType="flex-start"
    style={
      [
        {
          "alignItems": "flex-start",
          "alignSelf": "flex-start",
          "flexDirection": "column",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#0e3f6c",
            "fontFamily": "Avenir",
            "fontSize": 18,
            "fontWeight": "900",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
        ]
      }
    >
      <TextInput
        autoCapitalize="none"
        placeholder=""
        secureTextEntry={false}
        style={
          [
            {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 30,
              "marginBottom": 5,
            },
          ]
        }
      />
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#C2D6DC",
              "fontSize": 25,
            },
            undefined,
            {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignSelf": "stretch",
            "borderColor": "#C2D6DC",
            "borderWidth": 0.5,
            "flexDirection": "row",
            "marginBottom": 15,
          },
        ]
      }
    />
  </View>
</RCTSafeAreaView>
`;

exports[`Text Input Component Should work as expected to set text type to password to get snapshot 1`] = `
<RCTSafeAreaView
  style={
    [
      {},
    ]
  }
>
  <View
    alignType="flex-start"
    style={
      [
        {
          "alignItems": "flex-start",
          "alignSelf": "flex-start",
          "flexDirection": "column",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#0e3f6c",
            "fontFamily": "Avenir",
            "fontSize": 18,
            "fontWeight": "900",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
        ]
      }
    >
      <TextInput
        autoCapitalize="none"
        placeholder=""
        secureTextEntry={true}
        style={
          [
            {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 30,
              "marginBottom": 5,
            },
          ]
        }
      />
      <Text
        accessibilityLabel="MaterialsIconShowPasswordTestId"
        allowFontScaling={false}
        onPress={[Function]}
        selectable={false}
        style={
          [
            {
              "color": "#C2D6DC",
              "fontSize": 25,
            },
            undefined,
            {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
        testID="MaterialsIconShowPasswordTestId"
      >
        
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignSelf": "stretch",
            "borderColor": "#C2D6DC",
            "borderWidth": 0.5,
            "flexDirection": "row",
            "marginBottom": 15,
          },
        ]
      }
    />
  </View>
</RCTSafeAreaView>
`;
